lib/Sema/TypeCheckType.cpp

```diff
@@ -3453,6 +3472,13 @@ void TypeChecker::checkUnsupportedProtocolType(Decl *decl) {
       decl->getDeclContext()->isModuleScopeContext())
     return;
 
+  // Global variable declarations are okay to skip, because at the top level of script/repl
```

Comment:
No worries. Also don't feel like you have to get everything right 100%, because clearly it was broken before, and even if its not perfect we could check something in that makes it less broken :-)

The only case I'd be concerned about is if your changes accidentally made it so that we were not emitting a diagnostic in a case where we should be -- passing down an invalid existential to SILGen and IRGen will likely cause crashes.